### PR TITLE
Fix the state issue when client secret regenerates

### PR DESCRIPTION
**Pull request by ShanChathusanda93**
### Proposed changes in this pull request

* Fix https://github.com/wso2/product-is/issues/7712
* Added the state change when regenerating client secret after revoking.

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenManagementDAOImpl.java [hunk 1]
```diff
@@ -520,21 +520,31 @@ public void updateAppAndRevokeTokensAndAuthzCodes(String consumerKey, Properties
                     throw new IdentityOAuth2Exception("New Consumer Secret is not specified.");
                 }
 
+                if (properties.containsKey(OAuthConstants.OAUTH_APP_NEW_STATE)) {
+                    // update the consumer secret of the oauth app when the new app state is available.
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenManagementDAOImpl.java]: Start the comment with a capital letter. Check other places as well.

Comment (ShanChathusanda93) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenManagementDAOImpl.java]: Edited the comments.

**Comment by mefarazath** [comment]
Hi @ShanChathusanda93,

From what I saw we are trying to change the state to active whenever someone calls the regenerate client secret method.  Is that the intended final behaviour?

**Comment by ShanChathusanda93** [comment]
Hi @mefarazath ,

Yes that is.

**Pull request merged**